src/Entities/Boat.h

```diff
@@ -31,8 +31,27 @@ class cBoat :
 	virtual void HandleSpeedFromAttachee(float a_Forward, float a_Sideways) override;
 
 	cBoat(double a_X, double a_Y, double a_Z);
-} ;
 
+	int GetTimeSinceLastHit(void) const { return m_TimeSinceLastHit; }
```

Comment:
`TimeSince` is neither a good name nor a good measure.
What are the units? Why not call it `Ticks` instead to make it clear?
Why make a variable that needs updating each tick to maintain it? Make it `LastHitWorldAge`, specifying the world's age (in ticks) when the boat was last hit, so that it needs no updating, yet it conveys the same information.

Also, I don't see the code that actually does update this field.